core/controllers/reader.py

```diff
@@ -249,12 +249,53 @@ class StatsEventsHandler(base.BaseHandler):
 
     REQUIRE_PAYLOAD_CSRF_CHECK = False
 
+    def validate_aggregated_stats(self, aggregated_stats):
+        """Checks whether the aggregated stats dict has the correct keys.
+
+        Args:
+            aggregated_stats: dict. Dict comprising of aggregated stats.
+
+        Returns:
+            Bool. True if the dict is valid.
+        """
+        valid_stats = True
+        if 'num_starts' not in aggregated_stats:
+            valid_stats = False
```

Comment:
Just raise an exception with an appropriate error message, otherwise it'll be hard to tell what's wrong when debugging in production since you don't have a stacktrace. Ditto below. Catch the exception in the controller and log an error.

Reply:
@seanlip Are you suggesting we shouldn't fail this noisily?

Reply:
Probably not in a way that results in the learner getting an alert -- that would be distracting. It's fine if it appears as a console error in the frontend, though, or a logging.error() in the backend.

Reply:
Done.